src/detect-bytemath.c

```diff
@@ -0,0 +1,1284 @@
+    FAIL_IF_NOT(bmd->flags == flags);
+    FAIL_IF_NOT(bmd->endian == DETECT_BYTEMATH_ENDIAN_BIG);
+    FAIL_IF_NOT(bmd->base == DETECT_BYTEMATH_BASE_DEC);
+
+    if (bmd != NULL)
+        DetectByteMathFree(bmd);
+
+    PASS;
+}
+
+static int DetectByteMathParseTest13(void)
+{
+    uint8_t flags = DETECT_BYTEMATH_FLAG_STRING |
+                    DETECT_BYTEMATH_FLAG_RELATIVE |
+                    DETECT_BYTEMATH_FLAG_BITMASK;
+
+    DetectByteMathData *bmd = DetectByteMathParse("bytes 4, offset 2, oper +, "
+                                                  "rvalue 39, result bar, "
+                                                  "relative,  string dec, bitmask "
+                                                  "0x8f40", NULL);
+
+    FAIL_IF(bmd == NULL);
+    FAIL_IF_NOT(bmd->nbytes == 4);
+    FAIL_IF_NOT(bmd->offset == 2);
+    FAIL_IF_NOT(bmd->oper == DETECT_BYTEMATH_OPERATOR_PLUS);
+    FAIL_IF_NOT(bmd->rvalue == 39);
+    FAIL_IF_NOT(strcmp(bmd->result, "bar") == 0);
+    FAIL_IF_NOT(bmd->bitmask_val == 0x8f40);
+    FAIL_IF_NOT(bmd->bitmask_shift_count == 6);
+    FAIL_IF_NOT(bmd->flags == flags);
+    FAIL_IF_NOT(bmd->endian == DETECT_BYTEMATH_ENDIAN_BIG);
+    FAIL_IF_NOT(bmd->base == DETECT_BYTEMATH_BASE_DEC);
+
+    if (bmd != NULL)
+        DetectByteMathFree(bmd);
+
+    PASS;
+}
+
+
+static int DetectByteMathParseTest14(void)
+{
+    /* incomplete */
+    DetectByteMathData *bmd = DetectByteMathParse("bytes 4, offset 2, oper +,"
+                                                  "rvalue foo", NULL);
+
+    FAIL_IF(bmd != NULL);
+
+    if (bmd != NULL)
+        DetectByteMathFree(bmd);
+
+    PASS;
+}
+
+static int DetectByteMathParseTest15(void)
+{
+
+    /* incomplete */
+    DetectByteMathData *bmd = DetectByteMathParse("bytes 4, offset 2, oper +, "
+                                                  "result bar", NULL);
+
+    FAIL_IF(bmd != NULL);
+
+    if (bmd != NULL)
+        DetectByteMathFree(bmd);
+
+    PASS;
+}
+
+static int MyUTHPacketMatchSig(Packet *p, const char *sig);
+static int MyUTHPacketMatchSig(Packet *p, const char *sig)
+{
+    DecodeThreadVars dtv;
+    ThreadVars th_v;
+
+    DetectEngineThreadCtx *det_ctx = NULL;
+
+    memset(&dtv, 0, sizeof(DecodeThreadVars));
+    memset(&th_v, 0, sizeof(th_v));
+
+    DetectEngineCtx *de_ctx = DetectEngineCtxInit();
+    FAIL_IF(de_ctx == NULL);
+
+    de_ctx->flags |= DE_QUIET;
+
+    de_ctx->sig_list = SigInit(de_ctx, sig);
+    FAIL_IF(de_ctx->sig_list == NULL);
+
+    SigGroupBuild(de_ctx);
+    DetectEngineThreadCtxInit(&th_v, (void *)de_ctx, (void *)&det_ctx);
+
+    SigMatchSignatures(&th_v, de_ctx, det_ctx, p);
+    FAIL_IF(PacketAlertCheck(p, de_ctx->sig_list->id) == 1);
+
+    Signature *s = de_ctx->sig_list;
+    FAIL_IF(s->sm_lists_tail[DETECT_SM_LIST_PMATCH] == NULL);
+
+    SigMatch *sm = NULL;
+    sm = s->sm_lists[DETECT_SM_LIST_PMATCH];
+    FAIL_IF(sm->type != DETECT_CONTENT);
+
+    SigGroupCleanup(de_ctx);
```

Comment:
not needed, `DetectEngineCtxFree` calls both `SigGroupCleanup` and `SigCleanSignatures`